server/go/pkg/server/server.go

```
@@ -0,0 +1,295 @@
+
+	return s.Stream(stream)
+}
+
+type streamer struct {
+	handlerFactory
+	handler requestHandler
+	options *Options
+}
+
+type handlerFactory interface {
+	NewHandler() requestHandler
+}
+
+func (s *streamer) NewHandler() requestHandler {
+	if s.handlerFactory != nil {
+		return s.handlerFactory.NewHandler()
+	}
+
+	crypto := aead.NewAES256GCM()
+
+	sf := appencryption.NewSessionFactory(
+		&appencryption.Config{
+			Service: s.options.ServiceName,
+			Product: s.options.ProductID,
+			Policy: appencryption.NewCryptoPolicy(
+				appencryption.WithExpireAfterDuration(s.options.ExpireAfter),
+				appencryption.WithRevokeCheckInterval(s.options.CheckInterval),
+			),
+		},
+		NewMetastore(s.options),
+		NewKMS(s.options, crypto),
+		crypto,
+		appencryption.WithSecretFactory(new(memguard.SecretFactory)),
+		appencryption.WithMetrics(false),
+	)
+
+	return &defaultHandler{
+		sessionFactory: sf,
+	}
+}
+
+func NewMetastore(opts *Options) appencryption.Metastore {
+	switch opts.Metastore {
+	case "rdbms":
+		// TODO: support other databases
+		db, err := newMysql(opts.ConnectionString)
+		if err != nil {
+			panic(err)
+		}
+
+		return persistence.NewSQLMetastore(db)
+	case "dynamodb":
+		sess := awssession.Must(awssession.NewSessionWithOptions(awssession.Options{
+			SharedConfigState: awssession.SharedConfigEnable,
+		}))
+
+		return persistence.NewDynamoDBMetastore(sess)
+	default:
+		return persistence.NewMemoryMetastore()
+	}
+}
+
+func NewKMS(opts *Options, crypto appencryption.AEAD) appencryption.KeyManagementService {
+	if opts.KMS == "static" {
+		kms, err := kms.NewStatic("thisistotallynotsecretdonotuse!!", aead.NewAES256GCM())
+		if err != nil {
+			panic(err)
+		}
+
+		return kms
+	}
+
+	kms, err := kms.NewAWS(crypto, opts.PreferredRegion, opts.RegionMap)
+	if err != nil {
+		panic(err)
+	}
+
+	return kms
+}
+
+func (s *streamer) Stream(stream pb.AppEncryption_SessionServer) error {
+	defer func() {
+		if s.handler != nil {
+			s.handler.Close()
```

missed error check with `Close`

---

Ignoring the error on deferred `Close` is a common practice. Just look at this as a `defer s.handler.Close()` with extra steps.